hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -88,15 +89,15 @@ private static String getStrictlyLowerTimestamp(String timestamp) {
       }
     });
 
-    if (!beginInstantTime.equals(DEFAULT_BEGIN_TIMESTAMP)) {
+    if (missingCheckpointStrategy == MissingCheckpointStrategy.READ_LATEST) {
```

Comment:
if (checkpoint is not present) or  (checkpoint is present but checkpoint commit is not present in active timeline ) {
   SNAPSHOT_QUERY
} 
I think this logic is missing here

Reply:
its not required. if begin timestamp is empty we check the missing checkpoint strategy to determine the right values for begin timestamp and end timestamp. 
and later we just rely on the strategy to determine whether we want to make incr query or snapshot query.